src/components/Navbar/Navbar.tsx

```diff
@@ -3,52 +3,79 @@
 import styles from "./Navbar.module.css";
 import Image from "next/image";
 import ars_logo from "../../../public/assets/ARS_2_cropped.png";
-import { FaArrowRightLong } from "react-icons/fa6";
+import { FaArrowRight, FaBars } from "react-icons/fa";
 import { useEffect, useState } from "react";
-import { Link } from "react-scroll";
+import Link from "next/link";
 
 export default function Navbar() {
   const [logo, setLogo] = useState(false);
+  const [isMenuOpen, setIsMenuOpen] = useState(false); // Menu open state
+  const [isNavbarVisible, setIsNavbarVisible] = useState(true); // Navbar visibility
+  const [prevScrollPos, setPrevScrollPos] = useState(0); // Previous scroll position
+
+  // Handle scroll behavior
+  function handleScroll() {
+    const currentScrollPos = window.scrollY;
+
+    // Show/hide navbar based on scroll direction
+    if (prevScrollPos > currentScrollPos) {
+      setIsNavbarVisible(true); // Scrolling up
+    } else {
+      setIsNavbarVisible(false); // Scrolling down
+    }
 
-  function handleChange() {
     const htmlDoc = document.documentElement;
     const percentScrolled = (htmlDoc.scrollTop / htmlDoc.clientHeight) * 100;
 
-    percentScrolled > 85 ? setLogo(true) : setLogo(false);
+    setLogo(percentScrolled > 85); // Show/hide logo based on scroll
+    setPrevScrollPos(currentScrollPos); // Update previous scroll position
   }
 
   useEffect(() => {
-    window.addEventListener("scroll", handleChange);
-    window.addEventListener("resize", handleChange);
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [prevScrollPos]);
```

Comment:
Add `handleScroll` to the dependency array. I.E.: `[prevScrollPos, handleScroll]`.